### PR TITLE
Basic directory functionality

### PR DESCRIPTION
**Pull request by schnoberts1**
Hi, 

I want to use this tar-no-std to embed files with directories in them. I'm not interested in the directory entries per-se, just the directory pre-fixed filenames. I'm wondering if you're interested in me adding this functionality to your tar-no-std repo? If you are, here's an issue PR. I refactored the tests to remove some repetition and this branch has the same fix as my previous PR. 

Regarding this PR, I am very new to Rust (this is day 3 of me learning Rust) so I am not sure if what I've done is idiomatic or not. If you want me to add this functionality (or some extension) to your repo  I am more than happy to rework the PR given your feedback. 

Kind regards, 

Andy

### CODE REVIEW

File: examples/alloc_feature.rs [hunk 1]
```diff
@@ -21,9 +21,15 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 */
+
+#[cfg(not(feature = "alloc"))]
```
Comment (phip1611) [inline examples/alloc_feature.rs:25]: can you please tell me a little more about the build error that you try to fix?

Comment (schnoberts1) [reply examples/alloc_feature.rs:25]: Here's what I see at 382c10c217d9e239a9923ffc5b323634bba1140a (i.e. head on main at the merge of PR #5). 

```
(base) andy@ice tar-no-std % cargo test
   Compiling bitflags v2.4.0
   Compiling regex-syntax v0.7.4
   Compiling log v0.4.20
   Compiling humantime v2.1.0
   Compiling termcolor v1.2.0
   Compiling libc v0.2.147
   Compiling memchr v2.5.0
   Compiling tar-no-std v0.2.0 (/Users/andy/code/pico/3rdparty/tar-no-std)
   Compiling aho-corasick v1.0.4
   Compiling errno v0.3.2
   Compiling rustix v0.38.9
   Compiling regex-automata v0.3.6
   Compiling is-terminal v0.4.9
   Compiling regex v1.9.3
   Compiling env_logger v0.10.0
error[E0432]: unresolved import `tar_no_std::TarArchive`
  --> examples/alloc_feature.rs:24:5
   |
24 | use tar_no_std::TarArchive;
   |     ^^^^^^^^^^^^----------
   |     |           |
   |     |           help: a similar name exists in the module: `archive`
   |     no `TarArchive` in the root
   ```

**Comment by schnoberts1** [comment]
> Hi!
> 
> > I want to use this tar-no-std to embed files with directories in them. I'm not interested in the directory entries per-se, just the directory pre-fixed filenames. I'm wondering if you're interested in me adding this functionality to your tar-no-std repo? If you are, here's an issue PR. I refactored the tests to remove some repetition and this branch has the same fix as my previous PR.
> 
> Thanks for working on this.
> 
> > Regarding this PR, I am very new to Rust (this is day 3 of me learning Rust) so I am not sure if what I've done is idiomatic or not.
> 
> Nice! Welcome to Rust :)
> 
> Could you tell me a little about your use-case? So, I guess you are limited to `no_std`? @schnoberts1
> 
> From [#9 (comment)](https://github.com/phip1611/tar-no-std/issues/9#issue-1869585726), I found:
> 
> > I should explain why these issues are turning up. I am writing a very simple web server for a Raspberry Pico (to learn Rust) and I have embedded the files to be served in the binary as a tar file. rust-no-std is a nice simple library for parsing that file using an ArchiveRef but I'm encountering some compatibility issues as I test the edge cases related to tar file content.
> 
> I wonder: You are creating a standard Linux app, don't you?

The Raspberry Pico doesn't run an operating system so the code I am writing doesn't build with `std`. I am creating the tar files on a Mac using gnu tar (gtar) so they are standard tar files as you'd see on Linux.
